### ISSUE TITLE
Prepare endpoints to view all violent and property crime data 

### ISSUE THREAD

**Issue by LarryBafundo** [18F/crime-data-api#329]
As a user, I want to view all violent and all property crimes as a trend for a given location and timeframe. 

These perspectives should be selectable from the homepage and explorer views, as part of the offense drop-down menu.

**Comment by LarryBafundo** [comment]
@harrisj what's the status of this?

**Comment by harrisj** [comment]
Oh, I didn't know this was assigned to me? Can I get a little more detail about what these are? Let me tap in @cacraig too

**Comment by LarryBafundo** [comment]
@harrisj, i've updated the user story to better reflect what we want to do here. i believe that the FE team is waiting to for this endpoint to complete their implementation. 

cc: @jeremiak

**Comment by harrisj** [comment]
So this is an aggregate of several distinct NIBRS categories for property or violent crimes then? I think I need to confer with @cacraig. I apologize for not getting to this sooner, but in the future please assign this to me if you are waiting for me to do this.

**Comment by LarryBafundo** [comment]
@harrisj will do and no worries. 

let's talk about this tomorrow morning. i'm unsure of what we can support at the incident level and we may need to confer with @cacraig. ideally, the API would aggregate all violent and property crime for a given location and timeframe at the trend and incident details levels. but if we can't support the latter, we can make the case for it not making the MVP. so let's talk about what we can readily support and what we can't. however, i do know the FE team is waiting on this before they can wrap up their side of things.

**Comment by LarryBafundo** [comment]
@harrisj, @cacraig are we clear on what we need to do here, based on our conversation during sprint planning? 

we just want to be able to view trends (summary data) for all violent and or all property crime at the state and national levels. whenever all violet/property crime is selected we will not show accompanying incident data.  let me know if there are any questions.

**Comment by harrisj** [comment]
Okay, so what we need is the ability to pass a RET A crime category argument to the `/incidents/count` endpoint for grouping and such, right?

**Comment by LarryBafundo** [comment]
just to be clear, this is about being able to show a summary trend for all violent or property crime at the national or state level. it does not involve NIBRS data

**Comment by harrisj** [comment]
So, I have another question here. Right now the `incidents/count` endpoint returns records like this:

``` json
{
"actual": 1201,
"classification": "Violent",
"cleared": 0,
"juvenile_cleared": 0,
"offense": "Murder and Nonnegligent Homicide",
"offense_category": "Homicide",
"offense_code": "SUM_HOM",
"reported": 1201,
"unfounded": 0,
"year": 1960
}
```

Did you want the `violent-crimes` request to return a sum of all the subcrimes within the category (ie, like the above example but with offense of `Violent Crimes` or such). Or do you want this request to return individual records for each of the crimes within the broad grouping (so it would have homicide counts and rape counts or such).

**Comment by LarryBafundo** [comment]
@harrisj: when a user wants to view (summary) trends for "all violent crime", the API should return the aggregated counts for homicide, rape, robbery, and aggravated assault.  Similarly, "all property crime" should return the aggregate of arson, burglary, larceny theft, and motor vehicle theft. we should stick with the same offense categories we are using in the navigation panel, this is simply an aggregation of those crime types. let me know if any of this is unclear.

**Comment by harrisj** [comment]
Okay, it looks like we are going to need to modify the `reta_month_offense_subcat_summary` table to also generate counts for these broader categories. Right now we have this:

| classification   | offense                          |
|------------------|----------------------------------|
| Property         | Burglary                         |
| Property         | Motor Vehicle Theft              |
| Property         | <null>                           |
| Violent          | Manslaughter by Negligence       |
| Violent          | Murder and Nonnegligent Homicide |
| Violent          | Rape                             |
| Violent          | Robbery                          |
| Violent          | <null>                           |
| <null>           | <null>                           |

Which is kinda there, but it's missing arson and larceny theft for Property for instance and we also don't have Aggravated Assault in there for Violent crimes. I think we might need to rebuild this table to get the proper crimes in there. I think I will need the help of @cacraig on this one.

**Comment by harrisj** [comment]
Okay, I did a little more digging. It turns out I was looking at the wrong DB for that table above. Instead, it is

| classification   | offense                          |
|------------------|----------------------------------|
| Property         | Burglary                         |
| Property         | Larceny                          |
| Property         | Motor Vehicle Theft              |
| Violent          | Assault                          |
| Violent          | Manslaughter by Negligence       |
| Violent          | Murder and Nonnegligent Homicide |
| Violent          | Rape                             |
| Violent          | Robbery                          |


So, it looks like you can do something like `/incidents/count?classification=Violent` or `/incidents/count?classification=Property` and you'll get total aggregated trends. There are two caveats though to the data we have here:

1. Arson is not in this table. It's a separate report and I wonder if we can include it with the SRS data
2. The `reta_offense` table defines both `Rape` and `Legacy Rape`, but only the former is here. But I don't seem to see any cases where Legacy Rape is used in the database we have so...

**Comment by harrisj** [comment]
We don't have Arson data in there, but this seems to be feature-complete. I would like to close this out if it seems to work for you @jeremiak, with the plan to add Arson statistics in later as noted in https://github.com/18F/crime-data-explorer/issues/235

**Comment by harrisj** [comment]
Also, it looks like I was wrong about Legacy Rape. It's an offense subcategory, not a top-level offense type, and it is in there if you do a query like https://crime-data-api.fr.cloud.gov/incidents/count/?by=year&offense_subcat_code=SUM_RPE_FRC_LEG

**Comment by jeremiak** [comment]
I am totally in favor of closing this in favor per @harrisj's suggestion
